core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -735,6 +738,10 @@ private[spark] class TaskSetManager(
     maybeFinishTaskSet()
   }
 
+  private def shouldTaskFailureEventuallyFailJob(reason: TaskEndReason): Boolean = {
```

Comment:
can you add a java doc for this that clarifies what this means. In particular, why the ones particular instances you had below are special cases.